overlord/managers_test.go

```diff
@@ -3272,6 +3283,97 @@ version: 1.0
 	c.Assert(chg.Status(), Equals, state.DoingStatus, Commentf("install-snap change failed with: %v", chg.Err()))
 }
 
+func (s *mgrsSuite) TestInstallWithAssumesIsRefusedEarly(c *C) {
+	s.prereqSnapAssertions(c)
+
+	revno := "1"
+	snapYamlContent := `name: some-snap
+version: 1.0
+assumes: [something-that-is-not-provided]
+`
+	snapPath, _ := s.makeStoreTestSnap(c, snapYamlContent, revno)
+	s.serveSnap(snapPath, revno)
+
+	mockServer := s.mockStore(c)
+	defer mockServer.Close()
+
+	st := s.o.State()
+	st.Lock()
+	defer st.Unlock()
+
+	_, err := snapstate.Install(context.TODO(), st, "some-snap", nil, 0, snapstate.Flags{})
+	c.Assert(err, ErrorMatches, `snap "some-snap" assumes unsupported features: something-that-is-not-provided \(try to update snapd and refresh the core snap\)`)
```

Comment:
Isn't the "..and refresh the core snap" part of the message going to be confusing when core18/20 is in use?

Reply:
Ah, sorry, I didn't notice this is just adding tests and not changing the code. Ignore for now.

Reply:
Yes! I think it's something we want to fix too and just mention "base snap" or similar.